### PR TITLE
files to setup repo with actions

### PR DESCRIPTION
**Pull request by chongr**

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - '**.go'
+      - '.github/workflows/golangci-lint.yml'
+
+jobs:
+  golangci:
+    name: Format
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go_version: ['1.16', '1.18']
```
Comment (maroux) [inline .github/workflows/golangci-lint.yml]: 1.17 instead ?

File: .github/workflows/gotest.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - main
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.16', '1.17', '1.18']
```
Comment (maroux) [inline .github/workflows/gotest.yml:15]: Last two versions are fine but doesn't hurt I guess

**Review by maroux — APPROVED** [review]

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (chongr) [reply .github/workflows/golangci-lint.yml]: oh I can add 1.17, in `go.mod` im using 1.18 which I think is the most recent stable version?

Comment (maroux) [reply .github/workflows/golangci-lint.yml]: Yep 1.18 should be the main target now.

**Pull request merged**